### ISSUE TITLE
Legend marks when 0-value opacity

### ISSUE THREAD

**Issue by arvind** [vega/vega-lite#3010]
In the following spec, the legend marks disappear because the second unit spec has an `opacity` of `0`. 

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "width": 600,
  "height": 300,
  "data": {"url": "data/stocks.csv"},
  "layer": [
    {
      "mark": "line",
      "encoding": {
        "x": {"field": "date", "type": "temporal", "axis": null},
        "y": {"field": "price", "type": "quantitative"},
        "color": {"field": "symbol", "type": "nominal"}
      }
    },
    {
      "mark": "point",
      "encoding": {
        "x": {"field": "date", "type": "temporal", "axis": null},
        "y": {"field": "price", "type": "quantitative"},
        "color": {"field": "symbol", "type": "nominal"},
        "opacity": {"value": 0}
      }
    }
  ]
}
```

In the following spec, the opacity is set interactively but still produces 0-value legend mark opacities:

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "width": 600,
  "height": 300,
  "data": {"url": "data/stocks.csv"},
  "selection": {"brush": {"type": "interval"}},
  "mark": "point",
  "encoding": {
    "x": {"field": "date","type": "temporal","axis": null},
    "y": {"field": "price","type": "quantitative"},
    "color": {"field": "symbol","type": "nominal"},
    "opacity": {
      "value": 0,
      "condition": {"selection": "brush","value": 1}
    }
  }
}
```

Besides these examples, perhaps we should always set legend opacity to some non-0 value? Are there cases where an invisible legend mark is desirable?

**Comment by kanitw** [comment]
> Are there cases where an invisible legend mark is desirable?

Definitely not IMHO. 

I guess the question here is what opacity value to use is appropriate.  I think the max between the `condition.value` and `value` seems appropriate?

**Comment by domoritz** [comment]
Yes, I think the max is appropriate.

**Comment by kanitw** [comment]
Will be fixed in #3080
